resources/install.rb

```diff
@@ -57,7 +57,11 @@ def db_package
     when 'debian'
       'db-util'
     when 'rhel'
-      node['platform_version'].to_i >= 7 ? 'compat-db47' : 'db4-utils'
+      if node['platform'] == 'amazon'
+        'db4-utils'
+      else
+        node['platform_version'].to_i >= 7 ? 'compat-db47' : 'db4-utils'
```

Comment:
node['platform_version'].to_i >= 7 && !platform?('amazon') ? 'compat-db47' : 'db4-utils' would accomplish this without the if/else